lib/ansible/module_utils/basic.py

```diff
@@ -773,7 +773,8 @@ class AnsibleModule(object):
     def __init__(self, argument_spec, bypass_checks=False, no_log=False,
                  check_invalid_arguments=True, mutually_exclusive=None, required_together=None,
                  required_one_of=None, add_file_common_args=False, supports_check_mode=False,
-                 required_if=None):
+                 required_if=None, bypass_options_checks=False, options_mutually_exclusive=None,
+                 options_required_together=None, options_required_one_of=None, options_required_if=None):
```

Comment:
For multi level sub specs if some of the args are same across different level it might result in a false error, that's why avoided having a recursive call.